pandas/io/parsers/readers.py

```diff
@@ -268,7 +268,17 @@ class _read_shared(TypedDict, Generic[HashableT], total=False):
 skipfooter : int, default 0
     Number of lines at bottom of file to skip (Unsupported with ``engine='c'``).
 nrows : int, optional
-    Number of rows of file to read. Useful for reading pieces of large files.
+    Number of rows of data to read. Useful for reading pieces of large files.
+    The following rows are excluded from the count:
+
+    * The header row containing column names,
+    * Rows before the header row, if ``header=1`` or larger,
+    * Fully commented rows,
+    * Rows skipped with ``skiprows``,
+    * Skipped blank lines.
```

Comment:
The last 3 seem obvious to me, it might be better to remove them since the doc page of `read_csv` is quite long already.

Reply:
Should I just say that `nrows` limits the number of rows of the returned dataframe? I can't find any edge case where that is not true.

Reply:
Sounds good

Reply:
Ok I've updated that now.